core: round_robin to use acceptResolvedAddresses()

temawi opened the pull request:
Switching to new API to allow eventual deprecation of handleResolvedAddresses()

ejona86 reviewed (commented):
Why does this not check for empty addresses?

temawi commented:
@ejona86 It does not check and fail on empty address lists because they are actually supported. You can see a test exercising this scenario here: https://github.com/grpc/grpc-java/blob/7665d3850b9c3063372beae2fda4a604379bce22/core/src/test/java/io/grpc/util/RoundRobinLoadBalancerTest.java#L243

ejona86 commented:
I don't think it is actually supported. It reports TRANSIENT_FAILURE, but it [looks like](https://github.com/grpc/grpc-java/blob/2684e79f86f0ebc98a88f58d87434b6225b6896e/core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java#L185) it will queue all RPCs. It will essentially just hang, which is no good.

temawi commented:
> I don't think it is actually supported. It reports TRANSIENT_FAILURE, but it [looks like](https://github.com/grpc/grpc-java/blob/2684e79f86f0ebc98a88f58d87434b6225b6896e/core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java#L185) it will queue all RPCs. It will essentially just hang, which is no good.

Updated to check for empty addresses and I also fixed a test that was making wrong assumptions.

ejona86 commented on core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java:
```diff
@@ -126,6 +133,8 @@ public void onSubchannelState(ConnectivityStateInfo state) {
       shutdownSubchannel(removedSubchannel);
     }
+
+    return !resolvedAddresses.getAddresses().isEmpty();
```
This is guaranteed to return `true`, right, because of the check at the beginning of the method?
